tests/ui/extra_unused_type_parameters.rs

```diff
@@ -71,7 +79,23 @@ where
         .filter_map(move |(i, a)| if i == index { None } else { Some(a) })
 }
 
-fn unused_opaque<A, B>(dummy: impl Default) {}
+fn unused_opaque<A, B>(dummy: impl Default) {
+    unimplemented!()
+}
+
+mod unexported_trait_bounds {
+    mod private {
+        pub trait Private {}
+    }
+
+    fn priv_trait_bound<T: private::Private>() {
+        unimplemented!();
+    }
+
+    fn unused_with_priv_trait_bound<T: private::Private, U>() {
+        unimplemented!();
+    }
+}
```

Comment:
Maybe I missed it, but I think there is a test for a `pub` function missing (should not get linted).

Reply:
As far as I can tell, that relies on the config option, right? Does `uitest` try different values of config vars as listed in `clippy.toml`?

Reply:
Have a look in the `test/ui-toml` dir on how to test different config values.